backends/tofino/bf-asm/salu_inst.cpp

```diff
@@ -551,7 +551,7 @@ Instruction *AluOP::pass1(Table *tbl_, Table::Actions::Action *act) {
     if (k1 && (k1->value < Target::STATEFUL_ALU_CONST_MIN() ||
                k1->value > Target::STATEFUL_ALU_CONST_MAX())) {
         if (k1->value >= (INT64_C(1) << tbl->alu_size()) ||
-            k1->value < (INT64_C(~0u) << (tbl->alu_size() - 1))) {
+            k1->value < (~UINT64_C(0) << (tbl->alu_size() - 1))) {
```

Comment:
To do a signed comparison, you need to cast the value to a signed type (int64_t).  I would expect this to give a signed vs unsigned comparison warning (and will do an unsigned comparison, which is wrong.)